### PR TITLE
Landing page for extension

### LINKED ISSUE #143: Set up landing page for the browser extension.
Some feedback we got on Twitter about sending people to the blog post for information about the browser extension:

> I installed, but there are way too many clicks and too much text. I clicked `CC Search Browser Extension`. That takes me to GitHub where I can find the links to install on Firefox | Chrome | Opera if I look hard enough. Want to increase the number of installs? Send people to a page with big buttons that link directly to the installs for Firefox/Chrome/Opera. This is really easy to use. 99% of people don't need/want to read the help/background, but you could link to the help within the UI.

I think it would be good to create a simple static landing page using Vocabulary and hosted on GitHub Pages that just provided easy links to install the extension + a bit of background. We can use this as a homepage for the extension.

@makkoncept, I'm assigning to you to flesh it out further and make this issue ready for contribution. Let me know if you have questions.

### PR DESCRIPTION
**Pull request by techievivek**
Fixes #143 

**Description**
Added landing page for the CC browser extension based on the Vocabulary web design system.

**Other information**
Looking for feedback and help to improve it more.

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by kgodey** [comment]
@makkoncept please ensure that you get feedback from @panchovm (Francisco, our UX designer) before merging/deploying this.

**Comment by makkoncept** [comment]
@kgodey Okay.

**Comment by makkoncept** [comment]
@techievivek I have modified the code to use vocabulary header, footer, and Github corner. You can check out the updated webpage here https://makkoncept.github.io/ccsearch-browser-extension/docs/ 
- The footer currently does not have SVG icons. CC vocabulary team is working on updating the docs.
- You can experiment with the slide-in animation. 
- Also, I think the second version of the other features is better than the first one.

Push the changes that you want and after a final check, we can get this reviewed by Francisco.

**Comment by makkoncept** [comment]
Also, please make sure to follow the [code formatting section in INSTRUCTIONS.md](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/INSTRUCTIONS.md#code-formatting). This will keep the formatting consistent.

**Comment by techievivek** [comment]
@makkoncept I will go through the codes and will let you know also I will format code.

**Comment by techievivek** [comment]
@makkoncept I have added the animation and formatted the code, please have a look.

**Comment by techievivek** [comment]
@makkoncept I have added meta tags for social sharing and changed title of the document. Please have a look.

**Comment by makkoncept** [comment]
I have tried this locally and the slide-in animation is confusing and not conventional. It only triggers when hovering over the pictures.

ezgif com-video-to-gif

You can try to improve it, but I would recommend that we remove this animation altogether.

**Comment by techievivek** [comment]
@makkoncept Yes, I have added to slide when it get hovered using JavaScript, should I remove it completely?
Anything else we can improve upon.

**Comment by makkoncept** [comment]
@techievivek removing the animation would be good as now it seems that it would feel abrupt and unexpected to the user. I have also added a commit to the PR, so make sure to pull the changes also.

I also noticed that though the formatting of the `style.css` file was according to the prettier, it was not the case with `index.html`. Maybe your editor is using a different formatter for HTML documents or there may be some different settings(eg: wrapline) that take precedence over prettier. Can you please look into it because we have to update `INSTRUCTIONS.md` itself as I have not tested it by setting up the project from scratch on a different computer.

**Comment by techievivek** [comment]
@makkoncept Yes, I will check it out and update you. Also I have pulled the latest commit and removed animation all together.

**Comment by techievivek** [comment]
@makkoncept I have formatted the HTML file based on prettier config, can you verify if it's working?

**Comment by makkoncept** [comment]
Yes, now it's good. Can you tell what was the problem?

**Comment by techievivek** [comment]
@makkoncept I think issue was with my settings, my HTML files were recognized as "django-html" and hence formatting was not done properly. I had to change it to normal HTML file and then applied formatting and it worked.
Can we now ask Francisco for the review?

**Comment by techievivek** [comment]
@panchovm We have made the landing page for the CC search browser extension, could you please have a look at the design and verify if it as per CC standard and let us know if any change is required.
Thank you

**Comment by panchovm** [comment]
Hi all. Sorry for being late on this.

I checked the landing page, and overall it looks good. However, here some comments:

1. Based on what @makkoncept said above, I assume we will replace the header to match the vocabulary version. That means: the white version with product logo on the left and `explore CC` on the right. 
2. If that so, then I will use the `lightergray` color as background color for the first section; above **Feature**.
3. For the Feature section, I will middle-align the title and will use the `body-big` size for describing each sub-section (i.e. "Search through million of images and filter…")
4. Finally, the sub-item called **More features for more productivity** is duplicated. I assume that is a bug, right?

Applying those changes, we are perfect.

**Comment by makkoncept** [comment]
@panchovm Thanks for the review. But, I think that you were checking out the outdated version of the webpage. I am sorry, I didn't pushed the new changes on my fork. The latest version is here (https://makkoncept.github.io/ccsearch-browser-extension/).

**Comment by panchovm** [comment]
Perfect. I see the improvements in the final section. Anyway, the graphic suggestion is something that we can work.

**Comment by techievivek** [comment]
@panchovm so we will work on point 2 & 3 right, others are already incorporated.

**Comment by techievivek** [comment]
@panchovm @makkoncept I have edited the PR with the suggested changes, please have a look.

**Comment by techievivek** [comment]
@panchovm @makkoncept Here is the live version: http://techievivek.com/cc/

**Comment by panchovm** [comment]
It looks good. I like it. Two last comments though:
1. **Features** title, middle align.
2. Left-align text when the image is on the left; right-align text when the image is on the right.

One question, the title **ONE-CLICK DOWNLOAD WITH EASY ATTRIBUTION** looks like inside a smaller container compare with the following text. Why is that? It will look more harmonic if both containers have the same size.

**Comment by techievivek** [comment]
@panchovm I have added alignment to feature-text, and about the ONE-CLICK DOWNLOAD WITH EASY ATTRIBUTION, it is similar to other feature-headings.

File: docs/style.css
Comment (makkoncept) [inline docs/style.css]: The features section content does not touches the ends of the browser. This is good as any modern web page has this. To accomplish this, you have applied padding to individual elements or subsections of the feature section. You can improve this by wrapping all the feature section in container div and apply the padding to that. Then if you want to move any other element/section inside, then you can apply padding to that.

File: docs/style.css
Comment (makkoncept) [inline docs/style.css]: Instead of directly applying the rule to `h3`, make a utility class for this and add that class to the element.

File: docs/style.css
Comment (makkoncept) [inline docs/style.css]: You can remove these rules (`.navbar` and `.navbar-brand`). They are redundant.

**Review by makkoncept — CHANGES_REQUESTED** [review]
- All the feature descriptions are right aligned. Francisco recommended above that Left-align text when the image is on the left; right-align text when the image is on the right.

Screenshot from 2020-03-31 10-47-52

**Comment by makkoncept** [comment]
- Also, currently the webpage is not responsive. Can you work on that please.


ezgif com-video-to-gif (1)

**Comment by techievivek** [comment]
@makkoncept I will improve on all the above suggestions. Thanks for a thorough review.

**Comment by techievivek** [comment]
@makkoncept I have incorporated all changes as per your suggestions, could you please verify if everything is to the point.

**Review by makkoncept — CHANGES_REQUESTED** [review]
@techievivek I have checked and found that there are still some minor bugs left when the mobile responsiveness of the web page is concerned. Please refer the screenshots below:

Screenshot from 2020-03-31 21-31-47

- due to the `overflow-x: hidden` rule, the buttons and the extension content is getting cropped out. Check the screenshots for mobile screens *Iphone 5/SE* and *Galaxy 5S* below.

Screenshot from 2020-03-31 21-32-06
Screenshot from 2020-03-31 21-32-10

**Comment by techievivek** [comment]
@makkoncept But when I am removing overflow-x, it is creating some problem.
Should I remove extra padding or margin for the small screen? That might help and maybe reduce font and button size.

File: docs/style.css
Comment (makkoncept) [inline docs/style.css]: You can remove this class. This is overriding the vocabulary defaults and in this case the default size looks good.

**Before**

Screenshot from 2020-03-31 21-44-10 (1)

**After**

Screenshot from 2020-03-31 21-44-27

**Review by makkoncept — CHANGES_REQUESTED** [review]

**Comment by makkoncept** [comment]
@techievivek You can try that. I don't know about reducing the font-size, as it might create accessibility issues on a small screen.

**Comment by techievivek** [comment]
@makkoncept I will first try by removing/reducing margin and padding, I will update you once I find/stuck somewhere.

**Comment by techievivek** [comment]
@makkoncept Could you please verify now, I have added responsiveness for all small screen size.

**Comment by kgodey** [comment]
@dhruvkb could you review usage of Vocabulary in this PR?

**Comment by dhruvkb** [comment]
@techievivek @makkoncept @kgodey I'll go through it today.

File: docs/style.css
Comment (dhruvkb) [inline docs/style.css:21]: You have used a lot of `!important`. Was that really `important`? (pun intended)

Comment (dhruvkb) [reply docs/style.css:21]: You can use padding and margin classes provided by Vocabulary. See the Spacing documentation in the Storybook.

File: docs/style.css
Comment (dhruvkb) [inline docs/style.css:70]: Bulma provides zero padding and zero margin classes. Using them would be better than redefining them in your own CSS.

File: docs/loadassets.js
Comment (dhruvkb) [inline docs/loadassets.js:19]: You should make a function and use it to inject all necessary SVGs into the DOM. See the Usage page in Vocabulary for an example.

File: docs/style.css
Comment (dhruvkb) [inline docs/style.css:3]: Having to hide `overflow-x` on body is a sign that something's off in your styling. Fixing the cause of the overflow and fixing it would be better than painting over the problem.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: docs/style.css
Comment (techievivek) [reply docs/style.css:3]: This one is resolved once I added responsiveness to the site. I will send updated PR with this one removed.

File: docs/style.css
Comment (techievivek) [reply docs/style.css:15]: Thank you for sticking by here, by applying this color in the navbar certainly doesn't change the background. Also, I have faced similar issue with some other classes and hence had to create custom classes and add ```!important```

File: docs/index.html
Comment (techievivek) [reply docs/index.html:34]: @panchovm could you please confirm if the present header color is good or we should move to the one provided in header?

File: docs/loadassets.js
Comment (makkoncept) [reply docs/loadassets.js]: @techievivek I have updated this.

File: docs/style.css
Comment (techievivek) [reply docs/style.css:70]: @dhruvkb @makkoncept How can I can apply for this one as it will added only for small screen.
Should I use JavaScript?

Comment (dhruvkb) [reply docs/style.css:61]: [CSS Media Queries on W3Schools](https://www.w3schools.com/css/css_rwd_mediaqueries.asp)

Comment (techievivek) [reply docs/style.css:61]: I have added the same thing, but you are suggesting me to use class.

File: docs/style.css
Comment (techievivek) [reply docs/style.css:12]: @dhruvkb

File: docs/style.css
Comment (techievivek) [reply docs/style.css:61]: @dhruvkb

Comment (dhruvkb) [reply docs/style.css:61]: Oh, my bad. Sorry. This is fine as it is.

File: docs/style.css
Comment (dhruvkb) [reply docs/style.css:12]: I'm actually waiting to see @panchovm's feedback on the orange header. I'm not sure if he'll allow it.

Comment (dhruvkb) [reply docs/style.css:12]: In either case, increase the specificity using IDs, don't use `!important`.

File: docs/index.html
Comment (techievivek) [reply docs/index.html:34]: @panchovm Could you please verify this?

**Comment by techievivek** [comment]
@kgodey I haven't heard back from @panchovm on whether the above header can be used or not? Earlier when he reviewed the design, he didn't suggest to change header, should I proceed with same or wait?

**Comment by kgodey** [comment]
I'll defer to @panchovm, I messaged him on Slack.

**Comment by techievivek** [comment]
Thank you @kgodey

File: docs/index.html
Comment (panchovm) [reply docs/index.html:34]: As @dhruvkb said, the header must remain untouched. We want to reach design consistency across all CC sites.

**Comment by panchovm** [comment]
The header must remain as it is in Figma design; white header. Our goal is to reach a design consistency and header and footer are essential for this purpose.

File: docs/index.html
Comment (techievivek) [reply docs/index.html:34]: @panchovm So it should be one in Vocabulary, right.

**Comment by techievivek** [comment]
@panchovm Footer too should be exact same and no need to remove anythi

**Comment by panchovm** [comment]
In the case of Footer, the nav links can be hidden because there is no navigation in this site

**Comment by techievivek** [comment]
@dhruvkb I have added the header, also without using `!important` styles are not applied properly.
Can you please go through the code once and look at it.

**Comment by techievivek** [comment]
@makkoncept Can you look at the updated PR and give your feedback.

**Comment by makkoncept** [comment]
@techievivek I have changed one image because it is more close to the current design of the extension. The images are very outdated but we need not work on the new ones now as the design of the extension will change again after the new vocabulary version is integrated. 
Aside from this, I think that the heading "More features for more productivity" would look good if it's center aligned. Use a utility class for this. This will also remove the need for defining the class `feature-title` as the same utility could be used here too.

**Comment by techievivek** [comment]
@makkoncept I haven't gone through new design of the extension I will update the images. Also I will incorporate all the suggested changes.
Thanks for feedback.

**Comment by makkoncept** [comment]
@techievivek I think working on editing new images will not be worthwhile as the current design of the extension will also change significantly after vocabulary integration.

**Comment by techievivek** [comment]
@makkoncept We will work on rest of things and when images will be ready we will update it.

**Comment by techievivek** [comment]
@makkoncept I have updated the PR by adding center-align CSS utility.

**Comment by makkoncept** [comment]
@techievivek Okay :+1: It seems good to me. You can resolve the rest of the issues raised by Dhruv and then ask for a final review from him.

**Comment by techievivek** [comment]
@dhruvkb Can you please have a look at the updated PR, new header has been included also other suggestions have been incorporated.
Thank you

**Comment by dhruvkb** [comment]
@techievivek if it's not too much trouble, can you attach some screenshots? That'd be much more convenient than me having to set the code up myself.

**Comment by techievivek** [comment]
@dhruvkb Yes, I will soon add some screenshots and a working link too.

**Comment by techievivek** [comment]
@dhruvkb Below is a screenshot for the header part.
browser extension

And this is the live project link: http://techievivek.com/cc/

**Comment by annatuma** [comment]
This is starting to look good @techievivek thank you for your work on this page.

A couple of things to note:
1) I'm going to do a content review of the page. This isn't a blocker to merging, a separate PR can be opened when I've got that ticket ready with requested content changes. I expect they will be minor.

2) This page should live on CC Search. Understood that it's currently in the Browser Extension repo. Ideally we'll find a way for it to remain in the current repo, but be integrated as a part of the CC Search frontend. This means the same nav bar as CC Search, "Extension" being added as a navigation item (cc @panchovm @brenoferreira), and living on the ccsearch.creativecommons.org subdomain. @dhruvkb or anyone else, can you please look into if this is technically possible to do? Otherwise it unfortunately requires transfer between repos and a page rewrite, which isn't an ideal outcome.

**Comment by techievivek** [comment]
@annatuma Thank you for the feedback.

**Comment by dhruvkb** [comment]
If the page is supposed to fit in with the rest of the CC Search layout, then only the body of the page i.e. the `<main>` content, should be written here, so that it can be iframed into CC Search without the header, footer and other ornamentation. But that is not optimal because iframes are technically not part of the main page and come with several limitations that hurt UX.

In an ideal situation, pages should reside in repos where they belong and if the page is a part of CC Search, the best course of action is to just move it there and write it like any other Vue page.

**Comment by kgodey** [comment]
My concern with the page living in CC Search is that any updates/features/documentation changes would then need to be coordinated across two repos.

@makkoncept what do you think?

**Comment by makkoncept** [comment]
@kgodey Yes, this can be an issue specifically at the initial stages when we would be adding/removing to the webpage more frequently. 

I also understand the need for adding this page on CC Search as the extension is essentially a " plugin for CC Search" which promises a different User Experience. The users must know that they can also access CC licensed content using a cross-browser extension.

Considering the above two points, I think that adding just the link to the page in the navbar of CC Search would be ideal. That being said, if migrating this to cc-search repo looks like a good option in the long run, then it should be done in the initial stages as the migration would require rewriting the page in Vue.

**Comment by makkoncept** [comment]
@kgodey What should be the path forward? Given the changes and modifications that will be happening to the page, I think that we should keep it in this repo and add a link to it on CC Search. But if migrating it is a better option, then that's also fine.

**Comment by kgodey** [comment]
Checking with @annatuma, will get back to you.

**Comment by annatuma** [comment]
Fine to keep it in the repo for the time being. We can see how things go and migrate at a later date if it seems crucial to do so.

The compromise here will be a change to the header/including a link to CC Search. Attached is a screenshot of what we'd like to, but in brief:

- Narrower header with an external link to "Go to CC Search" (must look exactly as in screenshot), which will open in a new tab

- Updated lead section to include the CC Search logo

updates to header and lead section

**Comment by annatuma** [comment]
Finally (and apologies for the delay), here are some minor content changes required. I'm pasting in existing text, and what it should be updated with, for context. This is organized by page section.

**Existing**
CC Search Extension

A Cross-Browser Extension which lets you search and filter content in the public domain and under Creative Commons licenses.

Use on your favourite browser

A Cross-Browser Extension which lets you search and filter content in the public domain and under Creative Commons licenses. 

**Updated**
CC Search Extension

A cross-browser extension, which lets you search for and filter content that is under Creative Commons licenses.

Use on your favorite browser

**Existing**
SEARCH AND FILTER CC LICENSED CONTENT

Search through millions of CC-licensed Images and use filters to find the best content that meets your needs.

**Updated**
SEARCH AND FILTER CC-LICENSED CONTENT

Search through millions of CC-licensed images and use filters to find the content that best meets your needs.

**Existing**
ONE-CLICK DOWNLOAD WITH EASY ATTRIBUTION

Attributing the owner/creator of CC-licensed content found using the extension is easy with one-click attribution.

**Updated**
ONE-CLICK DOWNLOAD WITH EASY ATTRIBUTION

Crediting the creator of CC-licensed content found using the extension is easy with one-click attribution.

**Existing**
BOOKMARK YOUR FAVOURITE IMAGES

Bookmarking the images will save them in the extension. You can view and remove your bookmarks from the bookmarks section.

**Updated**
BOOKMARK YOUR FAVORITE IMAGES

Bookmarking images will save them in the extension. You can view and remove your bookmarks from the Bookmarks section.

**Existing**
DARK MODE

Dark Mode! Because everyone loves it. Good for your eyes and saves battery.

**Updated**
_No updates here, it's great._

Thank you again for jumping on this project @techievivek it's exciting to see this so close to going live.

**Comment by techievivek** [comment]
@annatuma Thank you for your inputs, I will update the PR and will let you know.

**Comment by techievivek** [comment]
@annatuma @makkoncept I have updated the PR in accordance with the above suggestion, please have a look
ccsearch
Also, the live version can be checked here: https://techievivek.com/cc

**Comment by techievivek** [comment]
@makkoncept @annatuma I have used font-awesome icon `<i class="fa fa-external-link" aria-hidden="true"></i>`, do we need to have credit as it is in [CC open-source website](https://opensource.creativecommons.org/)

**Comment by dhruvkb** [comment]
@techievivek you can use icons from [the Vocabulary set](https://cc-vocabulary.netlify.app/?path=/docs/tokens-icons--figma). You don't need to import Font Awesome at all.

**Comment by techievivek** [comment]
@dhruvkb Thank you.

**Comment by makkoncept** [comment]
@techievivek Nice work! It looks good to me. Waiting for final confirmation from @annatuma and then we can merge this. Thanks again for working on it.

**Comment by annatuma** [comment]
This is looking great @techievivek.

One thing, please swap out the CC logo with the CC Search logo. Sorry that wasn't clear from the screenshot.

(As a side note, we just got approval for an update to the CC Search logo, so that change will be made in Vocabulary soon. Until then, please use the product logo from the Design Library. )

**Comment by techievivek** [comment]
@annatuma I have updated the PR with the CC search logo. Please have a look.
cc search extension
The __live version__ can be viewed here: http://techievivek.com/cc/

**Comment by techievivek** [comment]
> @techievivek Nice work! It looks good to me. Waiting for final confirmation from @annatuma and then we can merge this. Thanks again for working on it.

Thank you for your guidance and for helping out whenever I get stuck.

**Comment by annatuma** [comment]
Looks great! Y'all rock 🥇

**Comment by panchovm** [comment]
@annatuma @techievivek The CC Search logo is available as SVG in [the Design Library](https://www.figma.com/file/l4Mt3dn3Ndtrvrb4aLcwXI/Design-Library?node-id=784%3A0&viewport=226%2C30%2C0.8688915371894836). Let me know if you can export the file.

**Comment by techievivek** [comment]
> @annatuma @techievivek The CC Search logo is available as SVG in [the Design Library](https://www.figma.com/file/l4Mt3dn3Ndtrvrb4aLcwXI/Design-Library?node-id=784%3A0&viewport=226%2C30%2C0.8688915371894836). Let me know if you can export the file.

I have exported the SVG version from the `assets` folder as described [here](https://cc-vocabulary.netlify.app/?path=/docs/assets-logos--cc-search)

**Review by makkoncept — APPROVED** [review]

**Comment by makkoncept** [comment]
The page is live on https://opensource.creativecommons.org/ccsearch-browser-extension/

**Comment by kgodey** [comment]
I updated the repo's URL to point to it. Nice work @techievivek and thanks for the help @dhruvkb

**Comment by techievivek** [comment]
@kgodey Thank you.

**Pull request merged**
